src/main/java/uk/gov/digital/ho/hocs/queue/common/QueueListener.java

```diff
@@ -0,0 +1,12 @@
+package uk.gov.digital.ho.hocs.queue.common;
+
+import org.springframework.messaging.handler.annotation.Header;
+
+import java.util.UUID;
+
+public interface QueueListener {
+    void onMessageReceived(String message,
+                           @Header("MessageId") String messageId,
+                           @Header(value = "MessageType", required = false) MessageType messageType,
```

Comment:
should message type be required?

Reply:
UKVI don't currently send this across - more for backwards compatibility at the minute. Ideally yes though...